src/main/drivers/vtx_smartaudio.c

```diff
@@ -0,0 +1,706 @@
+#include <stdbool.h>
+#include <stdint.h>
+#include <ctype.h>
+
+#include "platform.h"
+
+#ifdef VTX_SMARTAUDIO
+
+#include "drivers/system.h"
+#include "drivers/serial.h"
+#include "drivers/vtx_smartaudio.h"
+#include "io/serial.h"
+#include "fc/runtime_config.h"
+#include "config/config_master.h"
```

Comment:
Yes, you're right - none of this code should be in the driver, I should have spotted that.

So I'd move all code into the IO layer. However the comments about reducing interdependencies still apply - I would still make the changes suggested above.